benchmarks/Lua/storage.lua

```diff
@@ -39,15 +39,15 @@ end
 function storage:build_tree_depth (depth, random)
     self.count = self.count + 1
     if depth == 1 then
-        -- With Lua, an array cannot be pre-allocated.
+        -- With Lua, an array cannot be pre-allocated. (actually yes, you can)
         -- We just compute the size like in others languages.
         return {n = random:next() % 10 + 1}
     else
-        local arr = {n = 4}
-        for i = 1, 4 do
-            arr[i] = self:build_tree_depth(depth - 1, random)
-        end
-        return arr
+        return {n = 4,
+        self:build_tree_depth(depth - 1, random),
```

Comment:
Would this still work if the array is allocated with something like `{n = 4, nil, nil, nil, nil}` to avoid the code duplication?

Reply:
`{n = 4, nil, nil, nil, nil}`

"draws back" the performances from 

**~300ms to ~350ms**

While i'm about to say is purely subjective, Lua doesn't "actually" supports preallocation, you can fill it with nil values to prevent a ^2 reallocation when needed, when you don't know what the values will be.

My point is, fill it with `nil` is more like a hack than anything else.

I feel like using nils hack and later setting values inside with a loop is dirtier than the small code duplication.

Reply:
I'm testing `nil`s right now. I can say for sue it matters in PUC Lua, checking LuaJIT source code right now

Reply:
Ok, I don't know why does it speed up your benchmark, but according the LuaJIT `rehashtab` function the array is empty, so the size is shrinked to 0.
![image](https://user-images.githubusercontent.com/5685050/97766485-8b51cb00-1b27-11eb-80e6-f1c9387f5db4.png)
Here is `{nil,nil,nil,nil,nil,nil,nil,nil}`

![image](https://user-images.githubusercontent.com/5685050/97766495-999fe700-1b27-11eb-8b46-d546f5ebcb9e.png)
here is `{false, false, false, false, false}`

I have a theory it changes the asize but not the allocated array itself